veg2hab/mozaiek.py

```diff
@@ -93,76 +104,301 @@ def __str__(self):
 
-            if not self.alleen_zelfstandig:
-                self.keys.append((self.habtype, False, Kwaliteit.MATIG))
+            ElmID | habtype | kwaliteit | vegtypen | complexdeel_percentage | omringing_percentage
+            Er is een rij voor ieder complexdeel in de omliggende vlakken.
 
-    def check(self, habtype_percentage_dict: Dict) -> None:
-        requested_habtype_percentage = 0
-        for key in self.keys:
-            requested_habtype_percentage += habtype_percentage_dict.get(key, 0)
+        Vult ook de tegengekomen_kwal_vegtypen en mozk_perc_dict in.
+        """
+        assert all(
+            col in omringd_door.columns
+            for col in [
+                "ElmID",
+                "habtype",
+                "kwaliteit",
+                "vegtypen",
+                "complexdeel_percentage",
+                "omringing_percentage",
+            ]
+        ), "Not all expected columns found in omringd_door in mozaiekregel.check"
+
+        assert (
+            not self.ook_mozaiekvegetaties or self.kwalificerende_vegtypen is not None
+        ), "kwalificerende_vegtypen not set in mozaiekregel.check"
 
         threshold = (
             self.mozaiek_threshold
             if not self.ook_als_rand_langs
             else self.mozaiek_als_rand_threshold
         )
 
-        # Threshold is behaald, dus TRUE
-        if requested_habtype_percentage >= threshold:
+        self._vul_mozk_perc_dict(omringd_door)
+
+        (
+            omringing_kwal_vlakken,
+            omringing_HXXXX,
+            tegengekomen_kwal_vegtypen,
+        ) = self._bepaal_kwalificerende_en_HXXXX_omringing(omringd_door)
+
+        self._vul_tegengekomen_kwal_SBB_VvN(tegengekomen_kwal_vegtypen)
+
+        if omringing_kwal_vlakken >= threshold:
             self.cached_evaluation = MaybeBoolean.TRUE
             return
 
-        unknown_habtype_percentage = habtype_percentage_dict.get(
-            ("HXXXX", True, Kwaliteit.NVT),
-            0,
-        )
-        # Threshold kan nog behaald worden, dus POSTPONE
-        if requested_habtype_percentage + unknown_habtype_percentage >= threshold:
+        # Als de totale omringing van kwalificerende vlakken + HXXXX voldoende is, kan deze
+        # in volgende ronden nog TRUE worden, dus voor nu dan POSTPONE
+        if omringing_kwal_vlakken + omringing_HXXXX >= threshold:
             self.cached_evaluation = MaybeBoolean.POSTPONE
             return
 
-        # Threshold kan niet meer behaald worden, dus FALSE
         self.cached_evaluation = MaybeBoolean.FALSE
 
-    def __str__(self):
-        complete_string = ""
-        complete_string += f"{'(als rand langs)' if self.ook_als_rand_langs else ''} "
-        complete_string += (
-            f"{'zelfstndg' if self.alleen_zelfstandig else 'zelfstndg/mozk'} "
+    def _vul_mozk_perc_dict(self, omringd_door: pd.DataFrame) -> None:
```

Comment:
Ik heb een lichte voorkeur voor functies die gewoon de waarde teruggeven, i.p.v. de staat van de class aan te passen, maakt het iets makkelijker te testen. Maar is ook prima hoor.

Reply:
Ben ik met je eens, heb het hier zo gelaten omdat de assigments wat ingewikkelder zijn dan self.field = waarde, en ik wilde die logica uit check() houden.